homeassistant/components/image_processing/gocr.py

```diff
@@ -0,0 +1,150 @@
+"""
+Local optical character recognition processing using GOCR.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing.gocr/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.core import split_entity_id
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, ImageProcessingEntity, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_UNRECOGNIZED = 'unrecognized'
+CONF_CHARS = 'chars'
+CONF_EXTRA_ARGUMENTS = 'extra_arguments'
+CONF_HEIGHT = 'height'
+CONF_ROTATE = 'rotate'
+CONF_GOCR_BIN = 'gocr_bin'
+CONF_THRESHOLD = 'threshold'
+CONF_WIDTH = 'width'
+CONF_X_POS = 'x_position'
+CONF_Y_POS = 'y_position'
+CONF_NEGATE = 'negate'
+
+DEFAULT_BINARY = 'gocr'
+DEFAULT_UNRECOGNIZED = '_'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=''): cv.string,
+    vol.Optional(CONF_EXTRA_ARGUMENTS, default=''): cv.string,
+    vol.Optional(CONF_UNRECOGNIZED, default=DEFAULT_UNRECOGNIZED): cv.string,
+    vol.Optional(CONF_CHARS, default=''): cv.string,
+    vol.Optional(CONF_HEIGHT, default=0): cv.positive_int,
+    vol.Optional(CONF_GOCR_BIN, default=DEFAULT_BINARY): cv.string,
+    vol.Optional(CONF_THRESHOLD, default=0): cv.positive_int,
+    vol.Optional(CONF_ROTATE, default=0): cv.positive_int,
+    vol.Optional(CONF_WIDTH, default=0): cv.positive_int,
+    vol.Optional(CONF_X_POS, default=0): cv.positive_int,
+    vol.Optional(CONF_Y_POS, default=0): cv.positive_int,
```

Comment:
Ok so these values should not all have defaults set. Please use an option without a default to denote absence of a value.

Also, in this case `vol.Inclusive` would be useful because X&Y should be given together (as with width& height)

Reply:
I did that just now. I hope it is good.